apps/api/src/app/messages/usecases/remove-messages-by-transactionId/remove-messages-by-transactionId.usecase.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { buildFeedKey, buildMessageCountKey, InvalidateCacheService } from '@novu/application-generic';
+import { MessageEntity, MessageRepository } from '@novu/dal';
+import { RemoveMessagesByTransactionIdCommand } from './remove-messages-by-transactionId.command';
+
+@Injectable()
+export class RemoveMessagesByTransactionId {
+  constructor(private messageRepository: MessageRepository, private invalidateCache: InvalidateCacheService) {}
+
+  async execute(command: RemoveMessagesByTransactionIdCommand) {
+    const subscriberIds = await this.messageRepository.find(
+      {
+        transactionId: command.transactionId,
+        _environmentId: command.environmentId,
+        _organizationId: command.organizationId,
+        deleted: false,
+        ...(command.channel && { channel: command.channel }),
+      },
+      '_subscriberId'
+    );
+
+    if (subscriberIds.length === 0) {
+      throw new NotFoundException('Invalid transactionId or channel');
+    }
+
+    for (const subscriberId of subscriberIds) {
+      await this.invalidateCache.invalidateQuery({
+        key: buildFeedKey().invalidate({
+          subscriberId: subscriberId._subscriberId,
```

Comment:
I am not sure this one will work because we store by subscriberId and not _subscriberId, we would need either to fetch the subscriberId here or find another way to find it.